hooks/useSearchResultQuery.ts

```diff
@@ -1,19 +1,14 @@
+import { NoteSearchPreview } from "@/interfaces/noteSearchPreview";
+import { SearchQueryResult } from "@/interfaces/searchQueryResult";
 import useSWR from "swr";
-import { Note } from "@prisma/client";
-import {fetcher} from "@/lib/fetcher";
-import {NotePreview} from "@/interfaces/notePreview";
 
-export interface SearchQueryResult {
-    notes: NotePreview[];
-    error: Error | undefined;
-    isLoading: boolean;
-}
-
-//TODO: implement proper search query backend
 export function useSearchResultQuery(searchTerm: string): SearchQueryResult {
-    const { data, error, isLoading } = useSWR<NotePreview[]>(
-        "/api/notes",
+    const fetcher = (input: RequestInfo, init: RequestInit) => fetch(input, init).then((res) => res.json());
```

Comment:
There is no need to define a fetcher in useSearchResultQuery. It would be better to reuse the fetcher from `@/lib/fetcher`.

Reply:
done